EXAMEN MIGUEL ANGEL PEÑA Y SANTIAGO ORTEGA

In [1]:
import pandas as pd
import numpy as np

In [2]:
#Caracteristicas generales, seguridad social en salud y educacion

# Links a los conjuntos de datos
url_cgsse ='https://github.com/niconomist98/DataAnalyticsUQ/raw/refs/heads/main/Datos/GEIH2025/cgsse.CSV'
url_ocupados = 'https://github.com/niconomist98/DataAnalyticsUQ/raw/refs/heads/main/Datos/GEIH2025/Ocupados.CSV'
url_ft ='https://github.com/niconomist98/DataAnalyticsUQ/raw/refs/heads/main/Datos/GEIH2025/Fuerza%20de%20trabajo.CSV'
url_desocupados = 'https://github.com/niconomist98/DataAnalyticsUQ/raw/refs/heads/main/Datos/GEIH2025/No%20ocupados.CSV'


In [3]:
# CGSSE
df_cgsse = pd.read_csv(url_cgsse, encoding='latin-1', sep=';', 
                       low_memory=False)

#Ocupados
df_ocupados=pd.read_csv(url_ocupados,encoding='latin-1',sep=';',
                        low_memory=False)

#Fuerza de trabajo
df_ft=pd.read_csv(url_ft,
                  encoding='latin-1',sep=';',low_memory=False)
#No ocupados
df_d=pd.read_csv(url_desocupados,encoding='latin-1',
                           sep=';',low_memory=False)

In [4]:
df_cgsse

,PERIODO,MES,PER,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,REGIS,AREA,CLASE,...,P6170,P3041,P3042,P3042S1,P3042S2,P3043,P3043S1,P3038,P3039,POB_MAY18
0,20250626,6,2025,8238974,1,1,1,10,NaN,2,...,2.0,NaN,5.0,2.0,NaN,2.0,NaN,1.0,2.0,1.0
1,20250626,6,2025,8238974,1,2,1,10,NaN,2,...,2.0,NaN,5.0,1.0,NaN,1.0,NaN,2.0,1.0,1.0
2,20250626,6,2025,8238975,1,1,1,10,NaN,2,...,2.0,NaN,3.0,3.0,NaN,NaN,NaN,2.0,1.0,1.0
3,20250626,6,2025,8238975,1,2,1,10,NaN,2,...,2.0,NaN,1.0,0.0,NaN,NaN,NaN,2.0,1.0,1.0
4,20250626,6,2025,8238975,1,3,1,10,NaN,2,...,2.0,NaN,1.0,0.0,NaN,NaN,NaN,1.0,2.0,1.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
67851,20250626,6,2025,8283060,1,1,1,10,88.0,1,...,2.0,NaN,5.0,2.0,NaN,2.0,NaN,1.0,2.0,1.0
67852,20250626,6,2025,8283060,1,2,1,10,88.0,1,...,2.0,NaN,5.0,2.0,NaN,2.0,NaN,2.0,1.0,1.0
67853,20250626,6,2025,8283062,1,1,1,10,94.0,1,...,2.0,NaN,3.0,3.0,NaN,NaN,NaN,2.0,1.0,1.0
67854,20250626,6,2025,8283062,1,2,1,10,94.0,1,...,2.0,NaN,3.0,2.0,NaN,NaN,NaN,1.0,2.0,1.0


In [5]:
df_cgsse['FEX_C18'].sum()

np.float64(52045001.000009045)

In [6]:
df_ft

,PERIODO,MES,PER,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,REGIS,AREA,CLASE,...,P3362S8,P3362S7A1,P6300,P6310,P6310S1,P6320,P6330,P6340,P6350,P6351
0,20250626,6,2025,8238974,1,1,1,50,NaN,2,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,20250626,6,2025,8238974,1,2,1,50,NaN,2,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,20250626,6,2025,8238975,1,1,1,50,NaN,2,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,20250626,6,2025,8238975,1,2,1,50,NaN,2,...,NaN,NaN,2.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,20250626,6,2025,8238975,1,3,1,50,NaN,2,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
54445,20250626,6,2025,8283060,1,1,1,50,88.0,1,...,NaN,NaN,2.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
54446,20250626,6,2025,8283060,1,2,1,50,88.0,1,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
54447,20250626,6,2025,8283062,1,1,1,50,94.0,1,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
54448,20250626,6,2025,8283062,1,2,1,50,94.0,1,...,NaN,NaN,2.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [7]:
df_ocupados

,PERIODO,MES,PER,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,REGIS,AREA,CLASE,...,P1881,P1881S1,P1882,P7240,P7240S1,OCI,INGLABO,RAMA2D_R4,RAMA4D_R4,OFICIO_C8
0,20250626,6,2025,8238974,1,1,1,60,NaN,2,...,3,NaN,3.0,3,NaN,1,950000.0,1,113,6111
1,20250626,6,2025,8238974,1,2,1,60,NaN,2,...,3,NaN,4.0,5,NaN,1,NaN,1,113,6111
2,20250626,6,2025,8238975,1,1,1,60,NaN,2,...,3,NaN,10.0,3,NaN,1,700000.0,1,113,6114
3,20250626,6,2025,8238975,1,3,1,60,NaN,2,...,3,NaN,10.0,3,NaN,1,2000000.0,1,145,6122
4,20250626,6,2025,8238976,1,1,1,60,NaN,2,...,14,NaN,NaN,8,NaN,1,600000.0,1,122,6112
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
29701,20250626,6,2025,8283057,1,3,1,60,41.0,1,...,8,NaN,25.0,5,NaN,1,3800000.0,85,8544,2310
29702,20250625,6,2025,8283058,1,1,1,60,NaN,1,...,14,NaN,NaN,2,NaN,1,3000000.0,71,7111,2161
29703,20250625,6,2025,8283058,1,2,1,60,NaN,1,...,3,NaN,10.0,1,NaN,1,1500000.0,47,4752,1420
29704,20250626,6,2025,8283060,1,2,1,60,88.0,1,...,3,NaN,10.0,8,NaN,1,500000.0,56,5619,7512


In [8]:
df_d

,PERIODO,MES,PER,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,REGIS,AREA,CLASE,...,P7360,P9460,P1519,P1883,P7422,P7422S1,OFICIO1_C8,OFICIO2_C8,RAMA2D_D_R4,RAMA4D_D_R4
0,20250626,6,2025,8238975,1,2,1,70,NaN,2,...,1.0,2.0,2,NaN,2,NaN,.,6112,1.0,0123
1,20250626,6,2025,8238978,1,1,1,70,NaN,2,...,1.0,2.0,2,NaN,2,NaN,.,9211,1.0,0161
2,20250626,6,2025,8238980,1,2,1,70,NaN,2,...,NaN,NaN,2,NaN,2,NaN,.,.,NaN,.
3,20250626,6,2025,8238981,1,2,1,70,NaN,2,...,NaN,NaN,2,NaN,2,NaN,.,.,NaN,.
4,20250626,6,2025,8238983,1,2,1,70,NaN,2,...,NaN,NaN,2,NaN,2,NaN,.,.,NaN,.
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
24739,20250626,6,2025,8283056,1,1,1,70,27.0,1,...,10.0,2.0,3,NaN,2,NaN,.,2341,85.0,8530
24740,20250626,6,2025,8283059,1,1,1,70,76.0,1,...,6.0,2.0,2,NaN,2,NaN,5223,5113,55.0,5511
24741,20250626,6,2025,8283060,1,1,1,70,88.0,1,...,9.0,2.0,2,NaN,2,NaN,.,9334,47.0,4711
24742,20250626,6,2025,8283062,1,2,1,70,94.0,1,...,NaN,NaN,2,NaN,2,NaN,.,.,NaN,.


# PUNTO 1

In [9]:
PT =  df_cgsse['FEX_C18'].sum()
print(f' La población total de referencia para Colombia basado en el modulo cgsse de la GEIH y el factor de expansión es {PT}')

 La población total de referencia para Colombia basado en el modulo cgsse de la GEIH y el factor de expansión es 52045001.000009045


La poblacion total para colombia en el año 2025 fue de 52045001.000009045

In [57]:
# en edad de trabajar P6040 :  ¿cuántos años cumplidos tiene...?
df_cgsse[df_cgsse['P6040']>=15]['FEX_C18'].sum()

np.float64(40653416.00000409)

La poblacion que esta en edad edad de trabajar es de 40653416.00000409

In [61]:
# Población menor de 15 años
no_pet = df_cgsse[df_cgsse['P6040'] < 15]['FEX_C18'].sum()

print("Población que no está en edad de trabajar:", no_pet)

Población que no está en edad de trabajar: 11391585.000004958


la poblacion que no esta en edad de trabajar es 11391585.000004958

In [12]:
#FT: Pertenece a la PEA: .sum()si: 1
PEA=df_ft[(df_ft['FT']==1)]['FEX_C18'].sum()
print(f'Población Economicamente Activa {PEA}')

Población Economicamente Activa 25974846.22356785


el total de la poblacion que se encuentra economicamente activa es de 25974846.22356785

In [13]:
#FFT Pertenece a la PEI  si:1
PEI=df_ft[(df_ft['FFT']==1)]['FEX_C18'].sum()
print(f'Población Economicamente Inactiva {PEI}')

Población Economicamente Inactiva 14678569.776436238


la poblacion que se encuentra econonomicamente inactiva es de 14678569.776436238

In [14]:
#OCI : Ocupado ? 1:si
O=df_ocupados[(df_ocupados['OCI']==1)]['FEX_C18'].sum()
print(f'Población Ocupada {O}')

Población Ocupada 23752952.660255846


In [15]:
#OCI : Ocupado ? 1:si
D=df_d[(df_d['DSI']==1)]['FEX_C18'].sum()
print(f'Población No ocupada {D}')

Población No ocupada 2221893.563312005


la poblacion que no se encuentra ocupada fue 2221893.563312005

In [16]:
PET = df_cgsse[df_cgsse['P6040']>=15]['FEX_C18'].sum()

In [17]:
TO=O/PET*100
print(f' Tasa de ocupacion 2025 {round(TO,1)}')

 Tasa de ocupacion 2025 58.4


la tasa de ocupacion para 2025 es de 58.4%

In [18]:
TD=D/PEA*100
print(f' Tasa de desempleo 2025 : {round(TD,1)}')

 Tasa de desempleo 2025 : 8.6


la tasa de desempleo fue de 8.6%

# PUNTO 2

In [19]:
#Primero hay que anexar la p3271 a la tabla de desempleados
df_d = pd.merge(df_d, df_cgsse[['DIRECTORIO', 'SECUENCIA_P', 'ORDEN', 'P3271','P6040']], 
                        on=['DIRECTORIO', 'SECUENCIA_P', 'ORDEN'], how='left')

df_ft = pd.merge(df_ft, df_cgsse[['DIRECTORIO', 'SECUENCIA_P', 'ORDEN', 'P3271','P6040']], 
                        on=['DIRECTORIO', 'SECUENCIA_P', 'ORDEN'], how='left')

QUINDIO

In [20]:
# 1. Calcular cantidad de hombres desempleados en el quindío 
HDQ = df_d[(df_d['DSI']==1) & (df_d['DPTO']==63) & (df_d['P3271']==1)]['FEX_C18'].sum()
print(f'Hombres Desempleados en el Quindio: {HDQ}')

# 2. Calcular cantidad de hombres en la PEA del quindío
PEAHQ = df_ft[(df_ft['FT']==1) & (df_ft['DPTO']==63) & (df_ft['P3271']==1) ]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa de hombres en el quindío: {PEAHQ}')

# 3. Calcular la tasa de desempleo en hombres del quindío
TDHQ = HDQ/PEAHQ
print(f'Tasa de Desempleo en hombres del quindío: {TDHQ*100}%')

Hombres Desempleados en el Quindio: 17869.520960767004
Poblacion Economicamente Activa de hombres en el quindío: 174625.87195169798
Tasa de Desempleo en hombres del quindío: 10.233031773040919%


la tasa de hombres desempleados es de 10.233031773040919% para el Quindio

In [21]:
# 1. Calcular cantidad de MUJERES desempleadas en el quindío 
MDQ = df_d[(df_d['DSI']==1) & (df_d['DPTO']==63) & (df_d['P3271']==2)]['FEX_C18'].sum()
print(f'Mujeres Desempleadas en el Quindio: {MDQ}')

# 2. Calcular cantidad de mujeres en la PEA del quindío
PEAMQ = df_ft[(df_ft['FT']==1) & (df_ft['DPTO']==63) & (df_ft['P3271']==2) ]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa de mujeres en el quindío: {PEAMQ}')

# 3. Calcular la tasa de desempleo en mujeres del quindío
TDMQ = MDQ/PEAMQ
print(f'Tasa de Desempleo en mujeres del quindío: {TDMQ*100}%')

Mujeres Desempleadas en el Quindio: 10618.534945967001
Poblacion Economicamente Activa de mujeres en el quindío: 116885.879263648
Tasa de Desempleo en mujeres del quindío: 9.084531863781265%


las mujeres desempleadas en el Quindio fue de  9.084531863781265%,  

RISARALDA

In [22]:
# 1. Calcular cantidad de hombres desempleados 
HDQ = df_d[(df_d['DSI']==1) & (df_d['DPTO']==66) & (df_d['P3271']==1)]['FEX_C18'].sum()
print(f'Hombres Desempleados en el Risaralda: {HDQ}')

# 2. Calcular cantidad de hombres en la PEA
PEAHQ = df_ft[(df_ft['FT']==1) & (df_ft['DPTO']==66) & (df_ft['P3271']==1) ]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa de hombres en el Risaralda: {PEAHQ}')

# 3. Calcular la tasa de desempleo en hombres
TDHQ = HDQ/PEAHQ
print(f'Tasa de Desempleo en hombres del Risaralda: {TDHQ*100}%')

Hombres Desempleados en el Risaralda: 18000.60370715
Poblacion Economicamente Activa de hombres en el Risaralda: 296342.328368219
Tasa de Desempleo en hombres del Risaralda: 6.074260064793518%


tasa de dempleo para Hombres en Risaralda fue 6.074260064793518%

In [23]:
# 1. Calcular cantidad de MUJERES desempleadas 
MDQ = df_d[(df_d['DSI']==1) & (df_d['DPTO']==66) & (df_d['P3271']==2)]['FEX_C18'].sum()
print(f'Mujeres Desempleadas en el Risaralda: {MDQ}')

# 2. Calcular cantidad de mujeres en la PEA
PEAMQ = df_ft[(df_ft['FT']==1) & (df_ft['DPTO']==66) & (df_ft['P3271']==2) ]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa de mujeres en el Risaralda: {PEAMQ}')

# 3. Calcular la tasa de desempleo en mujeres
TDMQ = MDQ/PEAMQ
print(f'Tasa de Desempleo en mujeres del Risaralda: {TDMQ*100}%')

Mujeres Desempleadas en el Risaralda: 21676.799766712
Poblacion Economicamente Activa de mujeres en el Risaralda: 203157.903399127
Tasa de Desempleo en mujeres del Risaralda: 10.669926891362644%


la tasa de mujeres desempleadas para Risaralda fue 10.669926891362644%

CALDAS

In [24]:
# 1. Calcular cantidad de hombres desempleados 
HDQ = df_d[(df_d['DSI']==1) & (df_d['DPTO']==17) & (df_d['P3271']==1)]['FEX_C18'].sum()
print(f'Hombres Desempleados en el Caldas: {HDQ}')

# 2. Calcular cantidad de hombres en la PEA
PEAHQ = df_ft[(df_ft['FT']==1) & (df_ft['DPTO']==17) & (df_ft['P3271']==1) ]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa de hombres en el Caldas: {PEAHQ}')

# 3. Calcular la tasa de desempleo en hombres
TDHQ = HDQ/PEAHQ
print(f'Tasa de Desempleo en hombres del Caldas: {TDHQ*100}%')

Hombres Desempleados en el Caldas: 18226.160047418998
Poblacion Economicamente Activa de hombres en el Caldas: 326447.467769983
Tasa de Desempleo en hombres del Caldas: 5.5831831601957065%


In [25]:
# 1. Calcular cantidad de MUJERES desempleadas 
MDQ = df_d[(df_d['DSI']==1) & (df_d['DPTO']==17) & (df_d['P3271']==2)]['FEX_C18'].sum()
print(f'Mujeres Desempleadas en el Caldas: {MDQ}')

# 2. Calcular cantidad de mujeres en la PEA
PEAMQ = df_ft[(df_ft['FT']==1) & (df_ft['DPTO']==17) & (df_ft['P3271']==2) ]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa de mujeres en el Caldas: {PEAMQ}')

# 3. Calcular la tasa de desempleo en mujeres
TDMQ = MDQ/PEAMQ
print(f'Tasa de Desempleo en mujeres del Caldas: {TDMQ*100}%')

Mujeres Desempleadas en el Caldas: 17214.491449226003
Poblacion Economicamente Activa de mujeres en el Caldas: 224206.533775
Tasa de Desempleo en mujeres del Caldas: 7.677961547053494%


TOLIMA

In [26]:
# 1. Calcular cantidad de hombres desempleados 
HDQ = df_d[(df_d['DSI']==1) & (df_d['DPTO']==73) & (df_d['P3271']==1)]['FEX_C18'].sum()
print(f'Hombres Desempleados en el Tolima: {HDQ}')

# 2. Calcular cantidad de hombres en la PEA
PEAHQ = df_ft[(df_ft['FT']==1) & (df_ft['DPTO']==73) & (df_ft['P3271']==1) ]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa de hombres en el Tolima: {PEAHQ}')

# 3. Calcular la tasa de desempleo en hombres
TDHQ = HDQ/PEAHQ
print(f'Tasa de Desempleo en hombres del Tolima: {TDHQ*100}%')

Hombres Desempleados en el Tolima: 28255.96613253
Poblacion Economicamente Activa de hombres en el Tolima: 391325.28931057
Tasa de Desempleo en hombres del Tolima: 7.220582697916321%


In [27]:
# 1. Calcular cantidad de MUJERES desempleadas 
MDQ = df_d[(df_d['DSI']==1) & (df_d['DPTO']==73) & (df_d['P3271']==2)]['FEX_C18'].sum()
print(f'Mujeres Desempleadas en el Tolima: {MDQ}')

# 2. Calcular cantidad de mujeres en la PEA
PEAMQ = df_ft[(df_ft['FT']==1) & (df_ft['DPTO']==73) & (df_ft['P3271']==2) ]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa de mujeres en el Tolima: {PEAMQ}')

# 3. Calcular la tasa de desempleo en mujeres
TDMQ = MDQ/PEAMQ
print(f'Tasa de Desempleo en mujeres del Tolima: {TDMQ*100}%')

Mujeres Desempleadas en el Tolima: 52100.16756370999
Poblacion Economicamente Activa de mujeres en el Tolima: 271316.21571268
Tasa de Desempleo en mujeres del Tolima: 19.20274740190366%


# RESPUESTA 2

El departamento con la tasa con mayor desempleo para hombre es Quindio 10.233031773040919% y las mujeres el departamento de Tolima 19.20274740190366%

El departamento con la menor tasa de desempleo para hombres es Caldas con un 5.5831831601957065% y las mujeres el departamento de Caldas con un 7.677961547053494%

# PUNTO 3 

Quindio

In [28]:
#Segmento de 18 a 24 años de edad

# 1. Calcular cantidad personas desempleadas entre los 18 - 24 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=18) & (df_d['P6040']<=24) & (df_d['DPTO']==63)]['FEX_C18'].sum()
print(f'Personas Desempleadas en el quindío entre los 18 - 24 años: {D1824}')

# 2. Calcular cantidad personas en la PEA entre los 18 - 24
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=18) & (df_ft['P6040']<=24) & (df_ft['DPTO']==63)]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa en el quindío entre los 18 - 24 años: {PEA1824}')

# 3. Calcular la tasa de desempleo en personas entre los 18 - 24 del quindío
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo en personas entre los 18 - 24 del quindío: {TD1824*100}%')

Personas Desempleadas en el quindío entre los 18 - 24 años: 3819.2724449800003
Poblacion Economicamente Activa en el quindío entre los 18 - 24 años: 29912.663758397
Tasa de Desempleo en personas entre los 18 - 24 del quindío: 12.768078683423386%


In [29]:
#Segmento de 18 a 24 años de edad

# 1. Calcular cantidad personas desempleadas entre los 18 - 24 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=24) & (df_d['P6040']<=30) & (df_d['DPTO']==63)]['FEX_C18'].sum()
print(f'Personas Desempleadas en el quindío entre los 24 - 30 años: {D1824}')

# 2. Calcular cantidad personas en la PEA entre los 18 - 24
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=24) & (df_ft['P6040']<=30) & (df_ft['DPTO']==63)]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa en el quindío entre los 24 - 30 años: {PEA1824}')

# 3. Calcular la tasa de desempleo en personas entre los 18 - 24 del quindío
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo en personas entre los 24 - 30 del quindío: {TD1824*100}%')

Personas Desempleadas en el quindío entre los 24 - 30 años: 7008.068900358
Poblacion Economicamente Activa en el quindío entre los 24 - 30 años: 51117.425383347
Tasa de Desempleo en personas entre los 24 - 30 del quindío: 13.709745449427672%


In [30]:
#Segmento de 18 a 24 años de edad

# 1. Calcular cantidad personas desempleadas entre los 18 - 24 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=30) & (df_d['P6040']<=40) & (df_d['DPTO']==63)]['FEX_C18'].sum()
print(f'Personas Desempleadas en el quindío entre los 30 - 40 años: {D1824}')

# 2. Calcular cantidad personas en la PEA entre los 18 - 24
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=30) & (df_ft['P6040']<=40) & (df_ft['DPTO']==63)]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa en el quindío entre los 30 - 40 años: {PEA1824}')

# 3. Calcular la tasa de desempleo en personas entre los 18 - 24 del quindío
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo en personas entre los 30 - 40 del quindío: {TD1824*100}%')

Personas Desempleadas en el quindío entre los 30 - 40 años: 7015.512022579001
Poblacion Economicamente Activa en el quindío entre los 30 - 40 años: 72731.96687714598
Tasa de Desempleo en personas entre los 30 - 40 del quindío: 9.645706453159914%


In [31]:
#Segmento de 18 a 24 años de edad

# 1. Calcular cantidad personas desempleadas entre los 18 - 24 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=40) & (df_d['P6040']<=50) & (df_d['DPTO']==63)]['FEX_C18'].sum()
print(f'Personas Desempleadas en el quindío entre los 40 - 50 años: {D1824}')

# 2. Calcular cantidad personas en la PEA entre los 18 - 24
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=40) & (df_ft['P6040']<=50) & (df_ft['DPTO']==63)]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa en el quindío entre los 40 - 50 años: {PEA1824}')

# 3. Calcular la tasa de desempleo en personas entre los 18 - 24 del quindío
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo en personas entre los 40 - 50 del quindío: {TD1824*100}%')

Personas Desempleadas en el quindío entre los 40 - 50 años: 4416.562673204001
Poblacion Economicamente Activa en el quindío entre los 40 - 50 años: 70586.28939453498
Tasa de Desempleo en personas entre los 40 - 50 del quindío: 6.256969605695048%


In [32]:

# 1.
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=60) & (df_d['DPTO']==63)]['FEX_C18'].sum()
print(f'Personas Desempleadas en el quindío 60 en adelante: {D1824}')

# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=60) & (df_ft['DPTO']==63)]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa en el quindío 60 en adelante: {PEA1824}')

# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo en personas entre los 60 en adelante del quindío: {TD1824*100}%')

Personas Desempleadas en el quindío 60 en adelante: 4297.130465619
Poblacion Economicamente Activa en el quindío 60 en adelante: 41690.707895215004
Tasa de Desempleo en personas entre los 60 en adelante del quindío: 10.307165991086942%


Risaralda

In [33]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=18) & (df_d['P6040']<=24) & (df_d['DPTO']==66)]['FEX_C18'].sum()
print(f'Personas Desempleadas entre los 18 - 24: {D1824}')

# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=18) & (df_ft['P6040']<=24) & (df_ft['DPTO']==66)]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa entre los 18 - 24: {PEA1824}')

# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo en personas entre los 18 - 24: {TD1824*100}%')

Personas Desempleadas entre los 18 - 24: 9385.755003442
Poblacion Economicamente Activa entre los 18 - 24: 63454.922070486005
Tasa de Desempleo en personas entre los 18 - 24: 14.79121665773423%


In [34]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=24) & (df_d['P6040']<=30) & (df_d['DPTO']==66)]['FEX_C18'].sum()
print(f'Personas Desempleadas en el quindío entre los 18 - 24: {D1824}')

# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=24) & (df_ft['P6040']<=30) & (df_ft['DPTO']==66)]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa en el quindío entre los 18 - 24: {PEA1824}')

# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo en personas entre los 18 - 24: {TD1824*100}%')

Personas Desempleadas en el quindío entre los 18 - 24: 10504.69809363
Poblacion Economicamente Activa en el quindío entre los 18 - 24: 90684.88337600001
Tasa de Desempleo en personas entre los 18 - 24: 11.5837366742538%


In [35]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=30) & (df_d['P6040']<=40) & (df_d['DPTO']==66)]['FEX_C18'].sum()
print(f'Personas Desempleadas: {D1824}')

# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=30) & (df_ft['P6040']<=40) & (df_ft['DPTO']==66)]['FEX_C18'].sum()
print(f'Poblacion Economicamente Activa: {PEA1824}')

# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Personas Desempleadas: 9922.605545110002
Poblacion Economicamente Activa: 153899.63411428098
Tasa de Desempleo: 6.447452329705859%


In [36]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=40) & (df_d['P6040']<=50) & (df_d['DPTO']==66)]['FEX_C18'].sum()


# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=40) & (df_ft['P6040']<=50) & (df_ft['DPTO']==66)]['FEX_C18'].sum()


# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Tasa de Desempleo: 7.3356816001245955%


In [37]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=60) & (df_d['DPTO']==66)]['FEX_C18'].sum()


# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=60) & (df_ft['DPTO']==66)]['FEX_C18'].sum()


# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Tasa de Desempleo: 3.6398607318240614%


CALDAS

In [38]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=18) & (df_d['P6040']<=24) & (df_d['DPTO']==17)]['FEX_C18'].sum()


# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=18) & (df_ft['P6040']<=24) & (df_ft['DPTO']==17)]['FEX_C18'].sum()


# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Tasa de Desempleo: 9.344949005517167%


In [39]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=24) & (df_d['P6040']<=30) & (df_d['DPTO']==17)]['FEX_C18'].sum()


# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=24) & (df_ft['P6040']<=30) & (df_ft['DPTO']==17)]['FEX_C18'].sum()


# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Tasa de Desempleo: 6.442118267858205%


In [40]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=30) & (df_d['P6040']<=40) & (df_d['DPTO']==17)]['FEX_C18'].sum()


# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=30) & (df_ft['P6040']<=40) & (df_ft['DPTO']==17)]['FEX_C18'].sum()


# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Tasa de Desempleo: 5.719162060970324%


In [41]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=40) & (df_d['P6040']<=50) & (df_d['DPTO']==17)]['FEX_C18'].sum()


# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=40) & (df_ft['P6040']<=50) & (df_ft['DPTO']==17)]['FEX_C18'].sum()


# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Tasa de Desempleo: 5.123140786732679%


In [42]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=60) & (df_d['DPTO']==17)]['FEX_C18'].sum()


# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=60) & (df_ft['DPTO']==17)]['FEX_C18'].sum()


# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Tasa de Desempleo: 4.060989381499756%


Tolima

In [43]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=18) & (df_d['P6040']<=24) & (df_d['DPTO']==73)]['FEX_C18'].sum()


# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=18) & (df_ft['P6040']<=24) & (df_ft['DPTO']==73)]['FEX_C18'].sum()


# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Tasa de Desempleo: 19.817064699055017%


In [44]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=24) & (df_d['P6040']<=30) & (df_d['DPTO']==73)]['FEX_C18'].sum()


# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=24) & (df_ft['P6040']<=30) & (df_ft['DPTO']==73)]['FEX_C18'].sum()


# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Tasa de Desempleo: 13.882960839796077%


In [45]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=30) & (df_d['P6040']<=40) & (df_d['DPTO']==73)]['FEX_C18'].sum()


# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=30) & (df_ft['P6040']<=40) & (df_ft['DPTO']==73)]['FEX_C18'].sum()


# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Tasa de Desempleo: 12.247228061873246%


In [46]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=40) & (df_d['P6040']<=50) & (df_d['DPTO']==73)]['FEX_C18'].sum()


# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=40) & (df_ft['P6040']<=50) & (df_ft['DPTO']==73)]['FEX_C18'].sum()


# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Tasa de Desempleo: 8.641041526465289%


In [47]:
# 1. 
D1824 = df_d[(df_d['DSI']==1) & (df_d['P6040']>=60) & (df_d['DPTO']==73)]['FEX_C18'].sum()


# 2.
PEA1824 = df_ft[(df_ft['FT']==1) & (df_ft['P6040']>=60) & (df_ft['DPTO']==73)]['FEX_C18'].sum()


# 3.
TD1824 = D1824/PEA1824
print(f'Tasa de Desempleo: {TD1824*100}%')

Tasa de Desempleo: 9.458632182113549%


# RESPUESTA 3

para el Quindio la tasa de desempleo mayor fue entre las edades 30-40 13.709745449427672% y el menor es entre 40-50 con un 6.256969605695048%

para Risaralda la mayor entre 18-24 14.79121665773423% y la menor en los 60 en adelante 3.6398607318240614%

para caldas la mayor entre 18-24  9.344949005517167%  y la menor en los 60 en adelante 4.060989381499756%

para Tolima entre 18-24 19.817064699055017% y la menor entre 40-50 8.641041526465289%

# PUNTO 4 

Quindio

In [48]:
# 1. Calcular  desempleados en el quindío 
HDQ = df_d[(df_d['DSI']==1) & (df_d['P3271']==2) & (df_d['P6040']>=18) & (df_d['P6040']<=25) & (df_d['DPTO']==63)]['FEX_C18'].sum()


# 2.
PEAHQ = df_ft[(df_ft['FT']==1) & (df_ft['P3271']==2) &  (df_ft['P6040']>=18) & (df_ft['P6040']<=25) & (df_ft['DPTO']==63)]['FEX_C18'].sum()


# 3.
TDHQ = HDQ/PEAHQ
print(f'Tasa de Desempleo en mujeres del quindío: {TDHQ*100}%')

Tasa de Desempleo en mujeres del quindío: 7.468210722605588%


Risaralda

In [49]:
# 1. Calcular  desempleados en el quindío 
HDQ = df_d[(df_d['DSI']==1) & (df_d['P3271']==2) & (df_d['P6040']>=18) & (df_d['P6040']<=25) & (df_d['DPTO']==66)]['FEX_C18'].sum()


# 2.
PEAHQ = df_ft[(df_ft['FT']==1) & (df_ft['P3271']==2) &  (df_ft['P6040']>=18) & (df_ft['P6040']<=25) & (df_ft['DPTO']==66)]['FEX_C18'].sum()


# 3.
TDHQ = HDQ/PEAHQ
print(f'Tasa de Desempleo en mujeres del quindío: {TDHQ*100}%')

Tasa de Desempleo en mujeres del quindío: 22.992769133081296%


Tolima

In [50]:
# 1. Calcular  desempleados en el quindío 
HDQ = df_d[(df_d['DSI']==1) & (df_d['P3271']==2) & (df_d['P6040']>=18) & (df_d['P6040']<=25) & (df_d['DPTO']==73)]['FEX_C18'].sum()


# 2.
PEAHQ = df_ft[(df_ft['FT']==1) & (df_ft['P3271']==2) &  (df_ft['P6040']>=18) & (df_ft['P6040']<=25) & (df_ft['DPTO']==73)]['FEX_C18'].sum()


# 3.
TDHQ = HDQ/PEAHQ
print(f'Tasa de Desempleo en mujeres del quindío: {TDHQ*100}%')

Tasa de Desempleo en mujeres del quindío: 21.37807100891119%


Caldas

In [51]:
# 1. Calcular  desempleados en el quindío 
HDQ = df_d[(df_d['DSI']==1) & (df_d['P3271']==2) & (df_d['P6040']>=18) & (df_d['P6040']<=25) & (df_d['DPTO']==17)]['FEX_C18'].sum()


# 2.
PEAHQ = df_ft[(df_ft['FT']==1) & (df_ft['P3271']==2) &  (df_ft['P6040']>=18) & (df_ft['P6040']<=25) & (df_ft['DPTO']==17)]['FEX_C18'].sum()


# 3.
TDHQ = HDQ/PEAHQ
print(f'Tasa de Desempleo en mujeres del quindío: {TDHQ*100}%')

Tasa de Desempleo en mujeres del quindío: 14.50010240532998%


# RESPUESTA 4

Para mujeres en entre los 18 y 25 la menor tasa de desempleo fue para el quindio 7.468210722605588%

# 5. ¿Cuántos hombres y mujeres hay en edad de trabajar en el quindio ? ¿Qué porcentaje de la población total representan ?

In [52]:
# Población en edad de trabajar en Quindío (>= 15 años)
pet_quindio_hombres = df_cgsse[(df_cgsse['DPTO']==63) & (df_cgsse['P6040']>=15) & (df_cgsse['P3271']==1)]['FEX_C18'].sum()
pet_quindio_mujeres = df_cgsse[(df_cgsse['DPTO']==63) & (df_cgsse['P6040']>=15) & (df_cgsse['P3271']==2)]['FEX_C18'].sum()

# Población total del Quindío
poblacion_total_quindio = df_cgsse[df_cgsse['DPTO']==63]['FEX_C18'].sum()

# Porcentaje de hombres y mujeres en edad de trabajar sobre la población total del Quindío
porcentaje_hombres_pet_quindio = (pet_quindio_hombres / poblacion_total_quindio) * 100
porcentaje_mujeres_pet_quindio = (pet_quindio_mujeres / poblacion_total_quindio) * 100

print(f'Hombres en edad de trabajar en el Quindío: {pet_quindio_hombres:,.2f}')
print(f'Mujeres en edad de trabajar en el Quindío: {pet_quindio_mujeres:,.2f}')
print(f'Representan el {porcentaje_hombres_pet_quindio:,.2f}% de la población total del Quindío (Hombres)')
print(f'Representan el {porcentaje_mujeres_pet_quindio:,.2f}% de la población total del Quindío (Mujeres)')

Hombres en edad de trabajar en el Quindío: 247,037.51
Mujeres en edad de trabajar en el Quindío: 261,387.93
Representan el 40.96% de la población total del Quindío (Hombres)
Representan el 43.34% de la población total del Quindío (Mujeres)


# RESPUESTA 5

hombres en edad de trabajar 247,037.51, que representan el  40.96% de la poblacion del quindio, mujeres en edad de trabajar 261,387.93 representan el 43.34% 

#6 ¿Qué porcentaje de la población del Tolima está en edad de trabajar ? 

In [53]:
# Población en edad de trabajar en Tolima (>= 15 años)
pet_tolima = df_cgsse[(df_cgsse['DPTO']==73) & (df_cgsse['P6040']>=15)]['FEX_C18'].sum()

# Población total del Tolima
poblacion_total_tolima = df_cgsse[df_cgsse['DPTO']==73]['FEX_C18'].sum()

# Porcentaje de la población del Tolima en edad de trabajar
porcentaje_pet_tolima = (pet_tolima / poblacion_total_tolima) * 100

print(f'Población en edad de trabajar en el Tolima: {pet_tolima:,.2f}')
print(f'Población total del Tolima: {poblacion_total_tolima:,.2f}')
print(f'El {porcentaje_pet_tolima:,.2f}% de la población del Tolima está en edad de trabajar.')

Población en edad de trabajar en el Tolima: 1,087,359.08
Población total del Tolima: 1,411,965.04
El 77.01% de la población del Tolima está en edad de trabajar.


# RESPUESTA 6

El porcentaje total de poblacion del Tolima en edad de trabajar es de 43.34% 

#7 ¿ Qué porcentaje de la población de Risaralda es población económicamente inactiva ? 

In [54]:
# Población económicamente inactiva (PEI) en Risaralda
pei_risaralda = df_ft[(df_ft['DPTO']==66) & (df_ft['FFT']==1)]['FEX_C18'].sum()

# Población total de Risaralda
poblacion_total_risaralda = df_cgsse[df_cgsse['DPTO']==66]['FEX_C18'].sum()

# Porcentaje de la población de Risaralda que es PEI
porcentaje_pei_risaralda = (pei_risaralda / poblacion_total_risaralda) * 100

print(f'Población Económicamente Inactiva en Risaralda: {pei_risaralda:,.2f}')
print(f'Población total de Risaralda: {poblacion_total_risaralda:,.2f}')
print(f'El {porcentaje_pei_risaralda:,.2f}% de la población de Risaralda es económicamente inactiva.')

Población Económicamente Inactiva en Risaralda: 332,743.42
Población total de Risaralda: 1,025,240.94
El 32.46% de la población de Risaralda es económicamente inactiva.


# RESPUESTA 7

El porcentaje total de la poblacion economicamente inactiva para Risaralda es de 32.46% 

#8 ¿Cuántas personas hay en el Quindío en condición de desempleo entre los 18 y los 30 años ?

In [55]:
# Personas desempleadas en Quindío entre 18 y 30 años
desempleados_quindio_18_30 = df_d[(df_d['DPTO']==63) & (df_d['DSI']==1) & (df_d['P6040']>=18) & (df_d['P6040']<=30)]['FEX_C18'].sum()

print(f'Personas desempleadas en el Quindío entre 18 y 30 años: {desempleados_quindio_18_30:,.2f}')

Personas desempleadas en el Quindío entre 18 y 30 años: 9,479.63


# RESPUESTA 8

Las personas desempleadas en el departamento de Quindio entre los 18 y los 30 años es de 9,479.63

#9. ¿Cuántas personas hay ocupadas en el departamento del Tolima , cuyas edades están entre los 20 y los 25 años , qué porcentaje de la población total del Tolima representan ? 

In [56]:
import pandas as pd

# Define the URL for the 'Ocupados' dataset
url_ocupados = 'https://github.com/niconomist98/DataAnalyticsUQ/raw/refs/heads/main/Datos/GEIH2025/Ocupados.CSV'
# Define the URL for the 'CGSSE' dataset (needed for P6040 and AREA)
url_cgsse ='https://github.com/niconomist98/DataAnalyticsUQ/raw/refs/heads/main/Datos/GEIH2025/cgsse.CSV'

# Load DataFrames
df_cgsse = pd.read_csv(url_cgsse, encoding='latin-1', sep=';', low_memory=False)
df_ocupados = pd.read_csv(url_ocupados, encoding='latin-1', sep=';', low_memory=False)

# Prepare df_cgsse subset for merging, renaming 'AREA' to 'DPTO_TEMP' to avoid column collision
df_cgsse_for_merge = df_cgsse[['DIRECTORIO', 'SECUENCIA_P', 'ORDEN', 'P6040', 'AREA']].copy()
df_cgsse_for_merge.rename(columns={'AREA': 'DPTO_TEMP'}, inplace=True)

# Merge df_ocupados with the prepared df_cgsse subset to get P6040 (age) and DPTO_TEMP (department) columns
df_ocupados = pd.merge(df_ocupados, df_cgsse_for_merge,
                        on=['DIRECTORIO', 'SECUENCIA_P', 'ORDEN'], how='left')

# 1. Calcular cantidad de personas ocupadas en el Tolima entre 20 y 25 años
ocupados_tolima_20_25 = df_ocupados[(df_ocupados['OCI']==1) & 
                                   (df_ocupados['P6040']>=20) & 
                                   (df_ocupados['P6040']<=25) & 
                                   (df_ocupados['DPTO_TEMP']==73)]['FEX_C18'].sum()

print(f'Personas ocupadas en el Tolima entre los 20 y 25 años: {ocupados_tolima_20_25:,.2f}')

# 2. Calcular la población total del Tolima using df_cgsse's AREA column
poblacion_total_tolima = df_cgsse[df_cgsse['AREA']==73]['FEX_C18'].sum()

# 3. Calcular el porcentaje de personas ocupadas (20-25 años) sobre la población total del Tolima
porcentaje_ocupados_tolima = (ocupados_tolima_20_25 / poblacion_total_tolima) * 100

print(f'Representan el {porcentaje_ocupados_tolima:,.2f}% de la población total del Tolima.')

Personas ocupadas en el Tolima entre los 20 y 25 años: 23,575.55
Representan el 4.62% de la población total del Tolima.


# Respuesta 9

Hay 23575 personas ocupadas en el Tolima y el porcentaje de la poblacion ocupada entre las edades de 20 y 25 es de 4.62% 